### PR TITLE
Recipes App Submission

### PR DESCRIPTION
**Pull request by CRClothier**
- You should follow the layout of the wireframes provided.
- Login page and registration page:
  - Should be built with Devise.
- Food list:
  - Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
  - Should lead to a form that allows users to add new food.
- Recipes list:
  - Should display a list of recipes created by the logged-in user as in the wireframe.
  - Should lead to recipe details.
  - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
  - Should display a list of all public recipes ordered by newest as in the wireframe.
  - Should lead to recipe details.
- Recipe details:
  - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
  - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (only for a group that has less than 3 members):
  - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
  - Should count the total food items and total price of the missing food.
-Make sure there are no N+1 queries happening.
-Create a navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @CRClothier and @AlejandroNeira98,  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have the adequate instructions to run your project locally, in your Readme you specified `Create a .env file and copy the content from .env.example file. Assign the values to username and password according to your DB credentials`  kindly add the `.env.example` file  to your project 👍🏿 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<h2>Sign up</h2>
```
Comment (MahmoudBakr23) [inline app/views/devise/registrations/new.html.erb:1]: - [ ] Please make sure to fix the issue where a user has signed up for the first use of the app. See the error image down below!
Screenshot (1)

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
```
Comment (MahmoudBakr23) [inline app/views/devise/registrations/new.html.erb:3]: **Optional**
- I highly recommend styling the part of signing up and logging in on the app for a better look and consistency with the rest of the design.
Screenshot (2)

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# This file is copied to spec/ when you run 'rails generate rspec:install'
+require 'spec_helper'
```
Comment (MahmoudBakr23) [inline spec/rails_helper.rb:5]: - [x] For the RSpec tests, I see you might have missed that. So, please add unit and integration tests for your app's funcionalities with covering both negative and positive scenarios. 🙏

File: app/models/food.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class Food < ApplicationRecord
```
Comment (MahmoudBakr23) [inline app/models/food.rb:3]: **Optional**
- Well done with associations so far. However, i'd be much better for your models to write some validations for preventing bad users' behavior.

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello team😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done setting up the controllers and routes so far
- Great work handling the migrations and setting up your database
- Well done with the associations between your models
- Great work with the views
- Nicely done setting up devise gem
- Correct Linters
- Descriptive PR and README

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/rails_helper.rb (hunk 3 above)
Comment (CRClothier) [reply spec/rails_helper.rb:5]: Captura de Pantalla 2022-08-06 a la(s) 17 22 58

Tests added and running 👍

File: app/models/food.rb (hunk 4 above)
Comment (CRClothier) [reply app/models/food.rb:3]: Captura de Pantalla 2022-08-06 a la(s) 17 26 28

Model validations added 👍

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (CRClothier) [reply app/views/devise/registrations/new.html.erb:1]: Error fixed 👍

Comment (Deepakdanger) [reply app/views/devise/registrations/new.html.erb:1]: - [ ] I am still facing the same issue while signing up. Please fix this.

Screenshot from 2022-08-06 23-00-17

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @AlejandroNeira98 & @CRClothier, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
